dashboard/modules/metrics/metrics_head.py

```diff
@@ -275,3 +335,36 @@ def _format_prometheus_output(prom_data: Dict[str, Any]) -> Optional[TaskProgres
         return TaskProgress(**kwargs)
 
     return None
+
+
+def _format_prometheus_output_by_task_names(
+    prom_data: Dict[str, Any]
+) -> TaskProgressByTaskNameResponse:
+    """
+    Returns a list of task names with number of tasks for
+    each state with that task name.
+    """
+    task_map = {}
+
+    if prom_data["status"] == "success" and prom_data["data"]["resultType"] == "vector":
```

Comment:
Maybe I should've asked this last time. But is "data" and "status" guaranteed to be at the output? Should we have some check like `status in prom_data`? Also is this API stable from prometheus?

Reply:
yes this is a stable api from prometheus